### PR TITLE
Openssl 3 base images

### PR DESCRIPTION
**Pull request by tarnowsc**
### Desired Outcome

*Please describe the desired outcome for this PR.  Said another way, what was
the original request that resulted in these code changes?  Feel free to copy
this information from the connected issue.*

### Implemented Changes

*Describe how the desired outcome above has been achieved with this PR. In
particular, consider:*

- _What's changed? Why were these changes made?_
- _How should the reviewer approach this PR, especially if manual tests are required?_
- _Are there relevant screenshots you can add to the PR description?_

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue ID: [insert issue ID]

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: VERSION_APPLIANCE [hunk 1]
```diff
@@ -1 +1 @@
-5.0
+PR-57
```
Comment (hdabrowski) [inline VERSION_APPLIANCE:1]: will need to be reverted later

File: Dockerfile [hunk 2]
```diff
@@ -1,25 +1,32 @@
-FROM cyberark/ubuntu-ruby-fips:latest
+FROM registry.tld/cyberark/ubuntu-ruby-builder:22.04 as builder
```
Comment (hdabrowski) [inline Dockerfile:1]: will need to be changed back to docker.io

File: Dockerfile.ubi [hunk 3]
```diff
@@ -1,14 +1,25 @@
-# Conjur Base Image (UBI)
-FROM cyberark/ubi-ruby-fips:latest
+FROM registry.tld/cyberark/ubi-ruby-builder:ubi9 as builder
```
Comment (hdabrowski) [inline Dockerfile.ubi:1]: will need to be changed back to docker.io

File: package.sh [hunk 4]
```diff
@@ -8,14 +8,16 @@ chmod +x docker-debify
 docker run --rm \
   -v "$(pwd)":"$(pwd)" \
   --workdir "$(pwd)" \
-  cyberark/phusion-ruby-fips:latest \
-  sh -c "apt-get update -y && apt-get install -y git && bundle lock --update=conjur-api"
+  registry.tld/cyberark/ubuntu-ruby-builder:22.04 \
```
Comment (hdabrowski) [inline package.sh:11]: will need to be changed back to docker.io

File: VERSION_APPLIANCE (hunk 1 above)
Comment (marek-jakubowski) [reply VERSION_APPLIANCE:1]: After merging this PR and being able to build appliance from master.

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,12 +9,24 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Nothing should go in this section, please add to the latest unreleased version
   (and update the corresponding date), or add a new version.
 
+## [1.20.1] - 2023-07-31
+
+### Changed
+- Conjur will now use the new FIPS Base Images.
```
Comment (marek-jakubowski) [inline CHANGELOG.md:15]: Line 15, 16 and 17 could be deleted.

File: CHANGELOG.md [hunk 6]
```diff
@@ -9,12 +9,24 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+- Display FIPS Mode status in the UI (requires temporary fix for OpenSSL gem).
+  [cyberark/conjur#2874](https://github.com/cyberark/conjur/pull/2874)
+
 ## [1.20.0] - 2023-07-11
 
 ### Added
 - Telemetry support
   [cyberark/conjur#2854](https://github.com/cyberark/conjur/pull/2854)
 
+## [1.19.6] - 2023-07-05
```
Comment (marek-jakubowski) [inline CHANGELOG.md:28]: To be removed.

**Review by jvanderhoof — DISMISSED** [review]
Looks good.

File: dev/Dockerfile.dev [hunk 7]
```diff
@@ -1,4 +1,4 @@
-FROM cyberark/ubuntu-ruby-postgres-fips:latest
+FROM registry.tld/cyberark/ubuntu-ruby-postgres-fips:22.04
```
Comment (marek-jakubowski) [inline dev/Dockerfile.dev:1]: will need to be changed back to docker.io

File: VERSION_APPLIANCE (hunk 1 above)
Comment (marek-jakubowski) [inline VERSION_APPLIANCE:1]: Prepared #2949 to clean this up.

**Review by marek-jakubowski — APPROVED** [review]

**Pull request merged**